### DISCUSSION TITLE
Error occurs when countdown hits 0.10s (Countdown Clicker App)

### DISCUSSION THREAD

**Discussion by ndgsghdj** [sst-inc/help#10] [Q&A - iOS] [answered]
**iOS Session 3**
The code works perfectly fine until the countdown timer hits 0.10, in which the simulator will freeze, and Xcode will display this message:
<img width="526" alt="Screenshot 2021-07-27 at 8 51 28 PM" src="https://user-images.githubusercontent.com/75611248/127156536-ee559c1d-9abd-49bb-9b4a-c42fde4f05b9.png">
Next to the message would be a line of code that says "@main"
<img width="57" alt="Screenshot 2021-07-27 at 9 13 38 PM" src="https://user-images.githubusercontent.com/75611248/127159538-d676e059-c0f6-4f06-8476-6050c24061b8.png">

I had expected the app to display a new screen that shows the number of clicks and the clicks per second of the user instead of freezing and displaying this message in AppDelegate.swift.
I can't seem to find the cause of the error, I have compared the section of the code with the variable clicksPerSecond in the NewViewController.swift file with the section of the code in the tutorial code, and they seem to be identical.
I have tried to redo the IBOutlet concerned with the variable clicksPerSecond, and the error still occurs.
Code for NewViewController:
```
import UIKit

class NewViewController: UIViewController {
    
    var clicks: Int?
    var allocatedTime: Double?

    @IBOutlet weak var clicksLabel: UILabel!
    @IBOutlet weak var clicksPerSecondLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        clicksLabel.text = "You got \(clicks!) clicks"
        clicksPerSecondLabel.text = "Your clicks per second was \(String(format: "%.2f", Double(clicks!)/allocatedTime!))"
    }

```

**Comment by Joeispro99** [comment, marked as answer]
Where is your variable clicksPerSecond? Do you mind sending the source code of your other View Controller as well?

**Reply by ndgsghdj** [reply]
This is my code for the other view controller:
```
import UIKit

class ViewController: UIViewController {
    var counter = 0
    var timeLeft = 30.0
    var timer = Timer()
    
    @IBOutlet weak var counterLabel: UILabel!
    @IBOutlet weak var timeLeftLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        counterLabel.text = "\(counter)"
        timeLeftLabel.text = "Time Left: \(String(format: "%.2f", self.timeLeft))"
}
    
    override func viewDidAppear(_ animated: Bool) {
        self.timer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { (_) in
            // type code here
            self.timeLeft -= 0.1
            if self.timeLeft <= 0 {
                self.timer.invalidate() // invalidate timer so it doesn't go on and on, wastes memory, and also causes the view controller to be presented continuously
                self.timeLeft = 30.0
                self.performSegue(withIdentifier: "showResults", sender: nil) // perform segue
                self.timeLeft = 0.0
            }
            self.timeLeftLabel.text = "Time Left: \(String(format: "%.2f", self.timeLeft))s" // rounds off to 2dp so it looks cleaner
            print(self.timeLeft) // log the time left, we will change this later on
        }
    }
    
    
    @IBAction func buttonClicked(_ sender: Any) {
        newClick()
    }
    func newClick() {
        counter += 1
        counterLabel.text = "\(counter)"
        
    }
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "showResults" {
            let destinationViewController = segue.destination as! NewViewController
            destinationViewController.clicks = self.counter
            destinationViewController.allocatedTime = self.timeLeft
        }
    }
    
}
```

**Reply by Joeispro99** [reply]
Ok i found out the problem, send me the screenshot of the connections inspector of ur 2nd view controller label

**Reply by Joeispro99** [reply]
<img width="1440" alt="Screenshot 2021-07-28 at 8 18 08 AM" src="https://user-images.githubusercontent.com/65899074/127243786-f1a59537-721c-4c2d-95b8-1e9063c7eaf8.png">
e.g. like this

**Reply by ndgsghdj** [reply]
<img width="1398" alt="Screenshot 2021-07-28 at 6 16 18 PM" src="https://user-images.githubusercontent.com/75611248/127306090-95900a39-ff22-43b0-a82e-4d0afe55c2bf.png">

**Reply by Joeispro99** [reply]
![unnamed](https://user-images.githubusercontent.com/65899074/127316638-f36103ce-3558-407a-a3e3-d32c6660ba1c.png)
Click on the thing in the red square

**Reply by ndgsghdj** [reply]
<img width="264" alt="Screenshot 2021-07-28 at 10 05 58 PM" src="https://user-images.githubusercontent.com/75611248/127336387-b0979c80-9e11-4181-b3d2-adeabc7b9c8d.png">
This?

**Reply by Joeispro99** [reply]
Yes. Click on the cross for the clicksPerSecond, then your app should work
<img width="264" alt="127336387-b0979c80-9e11-4181-b3d2-adeabc7b9c8d" src="https://user-images.githubusercontent.com/65899074/127417034-d1578bf9-a153-43e4-bbdf-830f007eeb8c.png">

**Reply by ndgsghdj** [reply]
Ah okay, thanks
